### PR TITLE
xc7z support

### PR DESCRIPTION
**Pull request by kgugala**
This PR adds initial support for xc7z family

It is a WIP. Fuzzer 005 fails on calculating deltas for CLBs.

### CODE REVIEW

File: tools/gen_part_base_yaml.cc [hunk 1]
```diff
@@ -8,16 +8,25 @@
```
Comment (JohnDMcMaster) [inline tools/gen_part_base_yaml.cc]: This looks like a standalone patch. Can you break this out into its own PR?

**Comment by mithro** [comment]
Can you please add `make format`?

File: fuzzers/001-part-yaml/Makefile.specimen [hunk 2]
```diff
@@ -1,5 +1,11 @@
+ifeq (${XRAY_XC7Z}, 1)
```
Comment (JohnDMcMaster) [inline fuzzers/001-part-yaml/Makefile.specimen]: Why doesn't the existing py work for xc7z?

File: database/xc7z/settings.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Unsetting all previous XRAY env variables
```
Comment (JohnDMcMaster) [inline database/xc7z/settings.sh]: Other files are named such that this should be called database/zynq7/settings.sh

File: database/xc7z/settings.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+export XRAY_ROI_FRAMES="0x00000000:0xffffffff"
+export XRAY_XC7Z="1"
+
+# All CLB's in part, all BRAM's in part, all DSP's in part.
+export XRAY_ROI_TILEGRID="SLICE_X0Y0:SLICE_X43Y99 RAMB18_X0Y0:RAMB18_X2Y39 RAMB36_X0Y0:RAMB36_X2Y19 DSP48_X0Y0:DSP48_X1Y39"
+
+# These settings must remain in sync
+export XRAY_ROI="SLICE_X34Y0:SLICE_X43Y99 RAMB18_X2Y0:RAMB18_X2Y39 RAMB36_X2Y0:RAMB36_X2Y19 IOB_X0Y0:IOB_X0Y99"
+# Most of CMT X0Y2.
+export XRAY_ROI_GRID_X1="-1"
```
Comment (JohnDMcMaster) [inline database/xc7z/settings.sh]: This is set explicitly in other projects. We should also set this explicitly

File: database/xc7z/settings.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Unsetting all previous XRAY env variables
+unset ${!XRAY_@}
+
+export XRAY_DATABASE="xc7z"
```
Comment (JohnDMcMaster) [inline database/xc7z/settings.sh]: Similarly this should be zynq7

File: database/xc7z/settings.sh [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+export XRAY_PART="xc7z010clg400-1"
+export XRAY_ROI_FRAMES="0x00000000:0xffffffff"
+export XRAY_XC7Z="1"
```
Comment (JohnDMcMaster) [inline database/xc7z/settings.sh]: Do we need this? Usually we check value of $XRAY_DATABASE == zynq7

**Comment by kgugala** [comment]
Right now fuzzer 001 uses debug bitstream which is not supported for xc7z family.

BTW it seems we've found a better way to handle the xc7z devices. I'll write more details soon.

File: tools/gen_part_base_yaml.cc (hunk 1 above)
Comment (kc8apf) [reply tools/gen_part_base_yaml.cc]: This new option doesn't seem to have anything to do with generating part YAMLs and duplicates bittool's get_device_id action.

Comment (JohnDMcMaster) [reply tools/gen_part_base_yaml.cc]: Oh, I didn't read that carefully enough. Yeah, I'd agree this is probably out of scope

**Comment by kc8apf** [comment]
> 
> 
> Right now fuzzer 001 uses debug bitstream which is not supported for xc7z family.
> 
> BTW it seems we've found a better way to handle the xc7z devices. I'll write more details soon.

FWIW, if BITSTREAM.GENERAL.DEBUGBITSTREAM isn't available, BITSTREAM.GENERAL.PERFRAMECRC works as well.  I preferred DEBUGBITSTREAM because it has more documentation available. Both cause frames to be written to FDRI individually which is all that is necessary to determine the config memory layout of the device.

**Comment by kgugala** [comment]
Yep, we switched to PERFRAMECRC and it works better. I'll clean the code and issue a new PR
